### ISSUE TITLE
open FSharp.Core isn't marked as unused

### ISSUE THREAD

**Issue by auduchinok** [dotnet/fsharp#6975]
This namespace is implicitly added to all F# files and it is not accounted when checking symbols from that namespace in Unused Opens analysis logic.

<img width="273" alt="Screenshot 2019-06-11 at 11 00 43" src="https://user-images.githubusercontent.com/3923587/59254288-948f7e80-8c38-11e9-9d12-20fd0a3c9748.png">

**Comment by bender2k14** [comment]
Here is a MWE of this bug in text...
```F#
module Top
open FSharp.Core
let [<Literal>] x = 0
```

...and as a failing test in [this commit](https://github.com/bender2k14/fsharp/commit/a4797f5f57381f61153ef0dd7f0721bec5152233).

**Comment by abelbraaksma** [comment]
But it's this really a bug? The lib is used in this example through the attribute, and there a valid reasons you may have to open it explicitly, ie to prevent shadowing from other opens.

**Comment by cartermp** [comment]
FSharp.Core is _special_ yet again here. Typically the answer for shadowing is to fully qualify, but I guess I could see why someone would want to have it opened for a different scope. Given that I don't think this is a clear bug anymore.

**Comment by auduchinok** [comment]
In this particular case `x` `Literal` attribute isn't shadowed by anything else in scope and would be resolved without `open FSharp.Core`, so I think it's an analyzer bug.

And `FSharp.Core` is nothing _that_ special here. It just has several `AutoOpen` assembly-level attributes (which all F# assemblies are allowed to have), and one of which brings `Literal` into scope implicitly.

**Comment by bender2k14** [comment]
Excellent explanation @auduchinok.  I have confirmed this.  Specifically...

```F#
module Top
[<AutoOpen>]
module M1 =
  let v1 = ()
module M2 =
  open M1  // should be marked as unused but is not
  let v2 = v1
```

Tested with:
Operating system: Windows 10
.NET Runtime kind: .NET Core 3.1
Editing Tools: Visual Studio Enterprise 2019 version 16.7.1

I will review the existing tests and report back.

**Comment by bender2k14** [comment]
The following test is nearly the same and asserts that no `open` is unused.  However, if `open M.N` (on [line 699](https://github.com/dotnet/fsharp/blob/50e7f16806975c55c7a9ff65fcde410a8b427308/vsintegration/tests/UnitTests/UnusedOpensTests.fs#L699)) is removed, then the code still compiles and means the same thing.

https://github.com/dotnet/fsharp/blob/50e7f16806975c55c7a9ff65fcde410a8b427308/vsintegration/tests/UnitTests/UnusedOpensTests.fs#L692-L704

**Comment by abelbraaksma** [comment]
I agree to the points where it isn't related to `FSharp.Core`, as since that's treated specially, it'll be difficult to analyze, and requires to flow through all other opens to check whether something that's in scope was shadowed by any of the modules and types that are the transitive closure of `FSharp.Core`.

For `AutoOpen` on anything else than the defaults, this makes more sense, and analysis will be simpler by just excluding the in scope types of they belong to an `AutoOpen`.

(btw, I believe a similar bug exists with type aliases, where an open is marked as unused, where it actually is used).
